### PR TITLE
Fix the bad use of the $this variable. fix #288

### LINKED ISSUE #288: Uncaught exception in libraries\icms\file\MediaUploadHandler.php
```
Message: Using $this when not in object context
Code: 0
File: C:\xampp71\htdocs\unifya7\libraries\icms\file\MediaUploadHandler.php
Line: 691
Trace: 
#0 C:\xampp71\htdocs\unifya7\modules\downloads\class\DownloadHandler.php(55): icms_file_MediaUploadHandler::setErrors('Unknown filetyp...')
#1 C:\xampp71\htdocs\unifya7\modules\downloads\class\DownloadHandler.php(45): DownloadsDownloadHandler->checkMimeType()
#2 C:\xampp71\htdocs\unifya7\include\functions.php(1755): DownloadsDownloadHandler->__construct(Object(icms_db_legacy_PdoDatabase))
#3 C:\xampp71\htdocs\unifya7\include\functions.php(1011): icms_getModuleHandler('download', 'downloads', 'downloads', true)
#4 C:\xampp71\htdocs\unifya7\modules\downloads\icms_version.php(100): icms_getTablesArray('downloads', Array)
#5 C:\xampp71\htdocs\unifya7\libraries\icms\module\Object.php(284): include('C:\\xampp71\\htdo...')
#6 C:\xampp71\htdocs\unifya7\modules\system\admin\modules\modules.php(105): icms_module_Object->loadInfo('downloads', false)
#7 C:\xampp71\htdocs\unifya7\modules\system\admin\modules.php(67): xoops_module_list()
#8 C:\xampp71\htdocs\unifya7\modules\system\admin.php(117): include_once('C:\\xampp71\\htdo...')
#9 C:\xampp71\htdocs\unifya7\htdocs\index.php(114): require('C:\\xampp71\\htdo...')
#10 {main}
```

### PR DESCRIPTION
**Pull request by fiammybe**

### CODE REVIEW

File: libraries/icms/file/MediaUploadHandler.php [hunk 1]
```diff
@@ -526,7 +526,7 @@ public function upload($chmod = 0644) {
 		if (!self::checkMaxHeight()) {
 			return false;
 		}
-		if (count($this->errors) > 0) {
+		if (count($errors) > 0) {
```
Comment (MekDrop) [inline libraries/icms/file/MediaUploadHandler.php]: Scrutinizer says that here could be some problems. So, I'm not sure if this is a goo fix.

File: libraries/icms/file/MediaUploadHandler.php [hunk 2]
```diff
@@ -691,7 +691,7 @@ public function sanitizeMultipleExtensions() {
 	 * @param   string  $error
 	 */
 	public function setErrors($error) {
-		$this->errors[] = trim($error);
+		$errors[] = trim($error);
```
Comment (MekDrop) [inline libraries/icms/file/MediaUploadHandler.php]:
```
$errors was never initialized. Although not strictly required by PHP, it is generally a good practice to add $errors = array(); before regardless.
``` says scrutinizer

File: libraries/icms/file/MediaUploadHandler.php [hunk 3]
```diff
@@ -701,12 +701,12 @@ public function setErrors($error) {
 	 */
 	public function getErrors($ashtml = true) {
 		if (!$ashtml) {
-			return $this->errors;
+			return $errors;
```
Comment (MekDrop) [inline libraries/icms/file/MediaUploadHandler.php]: Another problem here.

File: libraries/icms/file/MediaUploadHandler.php (hunk 1 above)
Comment (fiammybe) [reply libraries/icms/file/MediaUploadHandler.php]: I looked at scrutinizer, and it look strange: they say that the private array $errors is not used, and after that, they say that the array is not defined. Perhaps there is a better way of handling this though (exceptions)

**Comment by fiammybe** [comment]
based on the feedback by @skenow that the error is not in our code, but due to a problem in the library that we are using, I will close this PR without merging.

**Pull request closed without merging**